data-products/src/main/scala/org/sunbird/obsrv/dataproducts/job/MasterDataProcessorIndexer.scala

```diff
@@ -0,0 +1,211 @@
+    } catch {
+      case e =>
+        metrics.generate(datasetId = dataset.id, edata = Edata(metric = Map(metrics.getMetricName("failure_dataset_count") -> 1), labels = List(MetricLabel("job", "MasterDataIndexer"), MetricLabel("datasetId", dataset.id), MetricLabel("cloud", "aws")), err = "Failed to index dataset.", errMsg = e.getMessage))
+    }
+  }
+
+  def getPaths(datasource: DataSource): Paths = {
+    val dt = new DateTime(DateTimeZone.UTC).withTimeAtStartOfDay()
+    val timestamp = dt.getMillis
+    val date = dayPeriodFormat.print(dt)
+    val cloudPrefix = if (config.getString("cloudStorage.provider") == "azure") {
+      getProviderUriFormat() + s"""${config.getString("cloudStorage.accountName")}.blob.core.windows.net/${config.getString("cloudStorage.container")}/"""
+    } else {
+      getProviderUriFormat() + s"""${config.getString("cloudStorage.container")}/"""
+    }
+    val pathSuffix = s"""masterdata-indexer/${datasource.datasetId}/$date/"""
+    val ingestionPath = cloudPrefix.replace(getProviderUriFormat(), getProviderFilePrefix()) + pathSuffix
+    val datasourceRef = datasource.datasource + '-' + date
+    val outputFilePath = cloudPrefix + pathSuffix
+    Paths(datasourceRef, ingestionPath, outputFilePath, timestamp)
+  }
+
+  private def updateIngestionSpec(datasource: DataSource, datasourceRef: String, filePath: String): String = {
+    val deltaIngestionSpec = s"""{"type":"index_parallel","spec":{"dataSchema":{"dataSource":"$datasourceRef"},"ioConfig":{"type":"index_parallel"},"tuningConfig":{"type":"index_parallel","maxRowsInMemory":500000,"forceExtendableShardSpecs":false,"logParseExceptions":true}}}"""
+    val provider = getProvider()
+    val inputSourceSpec = s"""{"spec":{"ioConfig":{"type":"index_parallel","inputSource":{"type":"$provider","objectGlob":"**.json.gz","prefixes":["$filePath"]}}}}"""
+    val deltaJson = parse(deltaIngestionSpec)
+    val inputSourceJson = parse(inputSourceSpec)
+    val ingestionSpec = parse(datasource.ingestionSpec)
+    val modIngestionSpec = ingestionSpec merge deltaJson merge inputSourceJson
+    compact(render(modIngestionSpec))
+  }
+
+  @throws[Exception]
+  private def getProviderUriFormat(): String = {
+    config.getString("cloudStorage.provider") match {
+      case "local" => "file:///home/sankethika/"
+      case "aws" => "s3a://"
+      case "azure" => "wasbs://"
+      case "gcloud" => "gs://"
+      case "cephs3" => "s3a://" // TODO: Have to check Druid compatibility
+      case "oci" => "s3a://" // TODO: Have to check Druid compatibility
+      case _ => throw new Exception("Unsupported provider")
+    }
+  }
+
+  @throws[Exception]
+  private def getProviderFilePrefix(): String = {
+    config.getString("cloudStorage.provider") match {
+      case "local" => "file"
```

Comment:
Please ref to this - https://github.com/Sunbird-Obsrv/sunbird-analytics-core/blob/release-4.8.0/analytics-core/src/main/scala/org/ekstep/analytics/framework/util/CommonUtil.scala#L761-L774

I don't think we are required to handle too many switch cases to get the `getProviderFilePrefix`, `getProviderUriFormat`, `getProvider` and `upload`

Reply:
Okay.